deploy/csi-azurefile-node-windows-hostprocess.yaml

```diff
@@ -0,0 +1,141 @@
+            - --health-port=29613
+            - --v=2
+          env:
+            - name: CSI_ENDPOINT
+              value: unix://C:\\var\\lib\\kubelet\\plugins\\file.csi.azure.com\\csi.sock
+        - name: node-driver-registrar
+          image: mcr.microsoft.com/oss/kubernetes-csi/csi-node-driver-registrar:v2.5.1
+          command:
+            - "csi-node-driver-registrar.exe"
+          args:
+            - "--v=2"
+            - "--csi-address=$(CSI_ENDPOINT)"
+            - "--kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)"
+            - "--plugin-registration-path=$(PLUGIN_REG_DIR)"
+          livenessProbe:
+            exec:
+              command:
+                - /csi-node-driver-registrar.exe
+                - --kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)
+                - --mode=kubelet-registration-probe
+            initialDelaySeconds: 60
+            timeoutSeconds: 30
+          env:
+            - name: CSI_ENDPOINT
+              value: unix://C:\\var\\lib\\kubelet\\plugins\\file.csi.azure.com\\csi.sock
+            - name: DRIVER_REG_SOCK_PATH
+              value: C:\\var\\lib\\kubelet\\plugins\\file.csi.azure.com\\csi.sock
+            - name: PLUGIN_REG_DIR
+              value: C:\\var\\lib\\kubelet\\plugins_registry\\
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+        - name: azurefile
+          image: mcr.microsoft.com/k8s/csi/azurefile-csi:latest
+          imagePullPolicy: Always
+          command:
+            - "azurefileplugin.exe"
+          args:
+            - --v=5
+            - --endpoint=$(CSI_ENDPOINT)
+            - --nodeid=$(KUBE_NODE_NAME)
+            - --kubeconfig=C:\\k\\config
+            - --metrics-address=0.0.0.0:29615
+            - --enableWindowsHostProcess=true
+          ports:
+            - containerPort: 29613
+              name: healthz
+              protocol: TCP
+          livenessProbe:
+            failureThreshold: 5
+            httpGet:
+              path: /healthz
+              port: healthz
+            initialDelaySeconds: 30
+            timeoutSeconds: 10
+            periodSeconds: 30
+          env:
+            - name: AZURE_CREDENTIAL_FILE
+              valueFrom:
+                configMapKeyRef:
+                  name: azure-cred-file
+                  key: path-windows
+                  optional: true
+            - name: CSI_ENDPOINT
+              value: unix://C:\\var\\lib\\kubelet\\plugins\\file.csi.azure.com\\csi.sock
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - name: pwsh
+              mountPath: "C:\\WindowsPowerShell"
+      volumes:
+        - name: pwsh
+          hostPath:
+            path: C:\Windows\System32\WindowsPowerShell\v1.0
+            type: DirectoryOrCreate
```

Comment:
@mauriciopoppe @alexander-ding How do you handle the powershell command inside container in your library solution? 
After testing, the powershell command cannot be found inside container in spite of using powershell related image. So here I mount powershell from host.

Reply:
I think the point is it's always using `C:\Windows\System32\WindowsPowerShell\v1.0`, and `v1.0` may change in the future, is there any way to avoid this though it works now?

Reply:
@ZeroMagic there's no need to mount anything. HostProcess containers have no file system isolation and have a complete view of the host machine's file system, so Poweshell should always be available at a well known path. There are a few options to expose powershell, besides what you're doing:
 
1. Expose the powershell folder in PATH. This is a bit ugly, but it should make things work.
2. Build your base image from the specialized [HostProcess container base image](https://github.com/microsoft/windows-host-process-containers-base-image) provided by Microsoft. This image is much smaller and works for all Windows server versions. Importantly, it makes powershell available directly as a command.

I'd recommend (2). This is what we're doing in the GCP Compute CSI Driver too.

Reply:
Will try it later.